test/crud_test.rb

```diff
@@ -101,6 +106,45 @@ class ModelUpdateOperation < CreateOperation
     ModelUpdateOperation[{id: 1, song: {title: "Mercy Day For Mr. Vengeance"}}].model.must_equal song
   end
 
+  # model Song, :action
+  class FetchCollectionOperation < CreateOperation
+    model Song
+
+    contract do
+      property :title
+    end
+
+    def setup_model!(params)
+      if @user_role == 'admin'
+        @collection = Song.all
+      else
+        @collection = nil
+      end
+    end
+
+    def setup_params!(params)
+      @user_role = "admin" if params[:user_id] == 0
```

Comment:
I'm setting @user_role here, idea is just differentiate on how fetch two different collection, could be:

```ruby
@myfoovar = true if params[:myfoovar] == 1
```

then if @myfoovar above